diffyscan/diffyscan.py

```diff
@@ -145,32 +211,127 @@ def run_diff(config, name, address, explorer_api_token, github_api_token, recurs
     logger.report_table(report)
 
 
-def process_config(path: str, recursive_parsing: bool, unify_formatting: bool):
+def process_config(
+    path: str,
+    recursive_parsing: bool,
+    unify_formatting: bool,
+    skip_binary_comparison: bool,
+):
     logger.info(f"Loading config {path}...")
     config = load_config(path)
 
-    github_api_token = load_env("GITHUB_API_TOKEN", masked=True)
     explorer_token = None
     if "explorer_token_env_var" in config:
-        explorer_token = load_env(config["explorer_token_env_var"], masked=True, required=False)
+        explorer_token = load_env(
+            config["explorer_token_env_var"], masked=True, required=False
+        )
+    if explorer_token is None:
+        logger.warn(
+            f'Failed to find an explorer token in the config ("explorer_token_env_var")'
+        )
+        explorer_token = os.getenv("ETHERSCAN_EXPLORER_TOKEN", default=None)
+    if explorer_token is None:
+        logger.warn(
+            f'Failed to find explorer token in env ("ETHERSCAN_EXPLORER_TOKEN")'
+        )
 
-    contracts = config["contracts"]
-    logger.info(f"Running diff for contracts from config {contracts}...")
-    for address, name in config["contracts"].items():
-        run_diff(config, name, address, explorer_token, github_api_token, recursive_parsing, unify_formatting)
+    if not skip_binary_comparison:
+        if "bytecode_comparison" not in config:
+            raise ValueError(f'Failed to find "bytecode_comparison" section in config')
+
+    github_api_token = os.getenv("GITHUB_API_TOKEN", "")
+    if not github_api_token:
+        raise ValueError("GITHUB_API_TOKEN variable is not set")
+
+    local_rpc_url = os.getenv("LOCAL_RPC_URL", "")
+    if not local_rpc_url:
+        raise ValueError("LOCAL_RPC_URL variable is not set")
+    if not local_rpc_url.startswith("http://"):
+        local_rpc_url = "http://" + local_rpc_url
```

Comment:
why do we do this :) It's a bit surprising and inconsistent when looking down at the `REPOTE_RPC_URL` case, e.g.

Reply:
fixed